weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx

```diff
@@ -477,30 +480,88 @@ const PanelPlotConfigInner: React.FC<PanelPlotProps> = props => {
             </ConfigPanel.ConfigOption>
           );
         })}
-        {config.series.map((series, i) => {
-          const seriesName = `Series ${i + 1} Name`;
+      </>
+    );
+  }, [config, updateConfig]);
+
+  const newSeriesConfigDom = useMemo(() => {
+    return (
+      <>
+        {config.series.map((s, i) => {
           return (
-            <ConfigPanel.ConfigOption
-              key={seriesName}
-              label={config.series.length > 1 ? seriesName : 'Series'}>
-              <ConfigPanel.TextInputConfigField
-                dataTest={`${seriesName}-label`}
-                value={series.seriesName}
-                label={''}
-                onChange={(event, {value}) => {
-                  updateConfig(
-                    produce(config, draft => {
-                      draft.series[i].seriesName = value;
-                    })
-                  );
-                }}
-              />
-            </ConfigPanel.ConfigOption>
+            <ConfigSection
+              label={`Series ${i + 1}`}
+              menuItems={[
+                {
+                  key: 'Remove series',
+                  content: 'Remove series',
+                  icon: <IconDelete />,
+                  onClick: () => {
+                    updateConfig(PlotState.removeSeries(config, s));
+                  },
+                },
+                {
+                  key: 'Add series from this series',
+                  content: 'Add series from this series',
+                  icon: <IconAddNew />,
+                  onClick: () => {
+                    const newConfig = PlotState.addSeriesFromSeries(
+                      config,
+                      s,
+                      'x',
+                      weave
+                    );
+                    updateConfig(newConfig);
```

Comment:
before  we call updateConfig here, we should set the new series name to something different from the original. maybe a default like "Series [N]" or something

Reply:
It already increments and creates "Series {N+1}" if there are currently N series.

Reply:
Thats the label for the config component, but not the actual series name, which lives in config.series[n].name

Reply:
I'm printing out the config object and I'm not seeing the name attribute on any series that I create, even on the first one. Where does it exist?

Reply:
Its on `config.series[n].name`

Reply:
https://github.com/wandb/weave/blob/master/weave-js/src/components/Panel2/PanelPlot/versions/v6.ts#L18

Reply:
added